test/spec/Feature/Query/RpcSpec.hs

```diff
@@ -1490,3 +1490,40 @@ spec actualPgVersion =
            `shouldRespondWith`
              [json| {"code":"22026","details":null,"hint":null,"message":"bit string length 6 does not match type bit(5)"} |]
            { matchStatus = 400 }
+
+    context "get message and details from raise sqlstate" $ do
+      it "gets message and details from raise sqlstate PGRST" $ do
+        get "/rpc/raise_sqlstate_test1" `shouldRespondWith`
+          [json|{"code":"123","message":"ABC","details":"DEF","hint":"XYZ"}|]
+          { matchStatus = 332
+          , matchHeaders = ["X-Header" <:> "str"] }
+
+        get "/rpc/raise_sqlstate_test2" `shouldRespondWith`
+          [json|{"code":"123","message":"ABC","details":null,"hint":null}|]
+          { matchStatus = 332
+          , matchHeaders = ["X-Header" <:> "str"] }
```

Comment:
Btw, we can also test the status message. There's an example on:

https://github.com/PostgREST/postgrest/blob/531a183b44b36614224fda432335cdaa356b4a0a/test/spec/Feature/Query/PlanSpec.hs#L26-L40

See `statusMessage="OK"`.